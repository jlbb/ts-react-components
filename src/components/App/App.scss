.app {
  font-family: 'IcoMoon-Free', 'sans-serif';
  max-width: 1600px;
  margin: 0 auto;
  //background-color: $backgroundColor;
}
