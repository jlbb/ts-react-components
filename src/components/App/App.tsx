import * as React from 'react';

type Props = typeof defaultProps;
type State = typeof defaultState;

const defaultProps = {};
const defaultState = {};

class App extends React.Component<Props, State> {
    static readonly defaultProps = defaultProps;
    readonly state = defaultState;

    render() {
        return (
            <div>
                <h1 className="title">Hola public World!</h1>
                <div className={'learning-icon'} />

                {this.props.children}
            </div>
        );
    }
}

export default App;
