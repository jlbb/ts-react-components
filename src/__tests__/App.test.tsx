import * as React from 'react';
import { render } from '@testing-library/react';
import App from '../components/App/App';

describe('My App default test case', () => {
    it('Should pass this test at any case', () => {
        expect(true).toEqual(true);
    });
});

describe('App', () => {
    test('snapshot without children renders', () => {
        const { container } = render(<App />);
        expect(container.firstChild).toMatchSnapshot();
    });
    test('snapshot with some children renders', () => {
        const { container } = render(
            <App>
                <h1>Hi World!</h1>
                <p>Lorem ipsum</p>
            </App>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
