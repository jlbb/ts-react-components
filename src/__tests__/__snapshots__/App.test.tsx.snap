// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot with some children renders 1`] = `
<div
  class="app"
>
  <h1>
    Hi World!
  </h1>
  <p>
    Lorem ipsum
  </p>
</div>
`;

exports[`App snapshot without children renders 1`] = `
<div
  class="app"
/>
`;
